### ISSUE TITLE
After update to 0.3.3 webui no longer works

### ISSUE THREAD

**Issue by Gozala** [ipfs/go-ipfs#1214]
ipfs webui fails to load additional assets:

```sh
curl -v http://localhost:5001/ipfs/QmXX7YRpU7nNBKfw75VG7Y1c3GwpSAGHRev67XVPgZFv9R/static/bundle.min.js
* Hostname was NOT found in DNS cache
*   Trying ::1...
* connect to ::1 port 5001 failed: Connection refused
*   Trying 127.0.0.1...
* Connected to localhost (127.0.0.1) port 5001 (#0)
> GET /ipfs/QmXX7YRpU7nNBKfw75VG7Y1c3GwpSAGHRev67XVPgZFv9R/static/bundle.min.js HTTP/1.1
> User-Agent: curl/7.37.1
> Host: localhost:5001
> Accept: */*
>
< HTTP/1.1 404 Not Found
< Date: Sat, 09 May 2015 01:14:27 GMT
< Content-Length: 102
< Content-Type: text/plain; charset=utf-8
<
* Connection #0 to host localhost left intact
Path Resolve error: no link named "bundle.min.js" under QmXX7YRpU7nNBKfw75VG7Y1c3GwpSAGHRev67XVPgZFv9R%

ipfs ls /ipfs/QmXX7YRpU7nNBKfw75VG7Y1c3GwpSAGHRev67XVPgZFv9R/static/                                                          
QmWHt1w9jB5fijBnFPpqN5s5HhEK4xAaboKG3h9FTB8kPa 1044019 bootstrap/
QmdFAD1HNhaptBhvx4Mso3tvWyA7zT9ojdVBpx2CGQkMd5 13721   bundle.min.css
Qmc2dngaxj9AGLWBRo4LntD3NssygqgdKsd7EcSXxKaPwR 968727  bundle.min.js
QmVXHyXW4MfXPcXNRmJRdt2waXKfo4BbPj2pjWnbYTgzxF 7107    css/
Qmc3nAUHEMUJmWEZnt8Vf4sC7vqsAuxxkrMzhNcvvRSb8E 771831  font-awesome/
Qmb3fJpXVGvUnNeRLC3P5sTXMzjpf5zq4tKt9XjhtYFf1k 216178  fonts/
QmQqcNFKHiSx5RPrhb7oFsQgn7DsrohuXAteqHRo1unqKb 199295  img/
QmRtuabZW95m3M2c7WPuEX2rMdiNkjxtWqvjU66BE4z2du 84408   jquery/
QmbHJUHDovCKoZQazGCFVrXGA9zuQJ7NokqD5vRYcYevDN 624206  js/
```

**Comment by Gozala** [comment]
Never mind, I was coping commands form the site, noticed I had to use a diff one ;)

**Comment by whyrusleeping** [comment]
Thanks for filing an issues, I think this has to do with changes made recently to the path resolution code. This very much implies a lack of test coverage on our part.

**Comment by wking** [comment]
On Fri, May 08, 2015 at 06:26:46PM -0700, Jeromy Johnson wrote:
> … I think this has to do with changes made recently to the path
> resolution code. This very much implies a lack of test coverage on
> our part.

Is this fixed by #1212?  That's currently waiting on me figuring out
how to add those tests, but maybe we should just merge it and follow
up with the tests in a new PR?

**Comment by whyrusleeping** [comment]
@wking that might be good, ill give #1212 a good CR in a minute and hopefully we can get the signoff from @jbenet tonight

**Comment by jbenet** [comment]
Did you mean to reopen?



—

**Comment by Gozala** [comment]
> Did you mean to reopen?

Yes, I just updated the description to reflect actual problem,  sorry for confusion (should have created separate issue instead).

**Comment by jbenet** [comment]
@Gozala no worries-- i missed @wking and @whyrusleeping's comments because my mail client separated the threads. (we're not hyper strict about messages, just making sure your issue gets addressed :) )

I think this is related to #1212 -- thanks for fixing that @wking. It's seeing whether this goes away-- @Gozala mind updating and testing? I'll try in a bit too.

**Comment by Gozala** [comment]
> I think this is related to #1212 -- thanks for fixing that @wking. It's seeing whether this goes away-- @Gozala mind updating and testing? I'll try in a bit too.

You mean try the master branch ? I'll give it a try later today and will post results.

**Comment by Gozala** [comment]
Attempting to load `http://localhost:5001/webui` now produces error messages:

```sh
➜  ~  go get -u github.com/ipfs/go-ipfs/cmd/ipfs
➜  ~  ipfs version
ipfs version 0.3.4
➜  ~  ipfs daemon
Initializing daemon...
API server listening on /ip4/127.0.0.1/tcp/5001
Gateway (readonly) server listening on /ip4/127.0.0.1/tcp/8080
13:09:34.763 ERROR core/serve: Path Resolve error: context deadline exceeded gateway_handler.go:436
13:09:34.770 ERROR core/serve: Path Resolve error: context deadline exceeded gateway_handler.go:436
13:09:34.770 ERROR core/serve: Path Resolve error: context deadline exceeded gateway_handler.go:436
13:09:34.770 ERROR core/serve: Path Resolve error: context deadline exceeded gateway_handler.go:436
13:09:34.771 ERROR core/serve: Path Resolve error: context deadline exceeded gateway_handler.go:436
13:09:34.771 ERROR core/serve: Path Resolve error: context deadline exceeded gateway_handler.go:436
^C
Received interrupt signal, shutting down...
13:09:48.872 ERROR core/serve: Path Resolve error: context canceled gateway_handler.go:436
13:09:48.872 ERROR core/serve: Path Resolve error: context canceled gateway_handler.go:436
13:09:48.872 ERROR core/serve: Path Resolve error: context canceled gateway_handler.go:436
13:09:48.873 ERROR core/serve: Path Resolve error: context canceled gateway_handler.go:436
13:09:48.873 ERROR core/serve: Path Resolve error: context canceled gateway_handler.go:436
13:09:48.873 ERROR core/serve: Path Resolve error: context canceled gateway_handler.go:436
13:09:48.878 ERROR core/serve: Path Resolve error: bitswap is closed gateway_handler.go:436
Publishing!
13:09:48.902 ERROR     ipnsfs: leveldb: closed system.go:83
➜  ~
```

I'm afraid I still see errors and webui is still not loading.

**Comment by jbenet** [comment]
Can anyone else repro this? i'm not seeing it.

@Gozala what's your `ipfs swarm peers` say?

**Comment by jbenet** [comment]
cc @krl @travisperson

**Comment by cryptix** [comment]
Tried as well, can't repo. 
The timeout also sounds like "no peers" to me, so yea. `ipfs swarm peers` please.

Maybe run with `IPFS_LOGGING=debug ipfs daemon` for a complete log.

**Comment by MrChrisJ** [comment]
I have had a similar issue attempting to load a file sent from a colleague running ipfs version 0.3.4, got the same error:

```
11:52:09.465 ERROR core/serve: Path Resolve error: context deadline exceeded gateway_handler.go:436
11:52:38.870 ERROR core/serve: Path Resolve error: context deadline exceeded gateway_handler.go:436
11:53:09.466 ERROR core/serve: Path Resolve error: context deadline exceeded gateway_handler.go:436
11:53:38.872 ERROR core/serve: Path Resolve error: context deadline exceeded gateway_handler.go:436
11:55:48.512 ERROR core/serve: Path Resolve error: context deadline exceeded gateway_handler.go:436
11:56:11.774 ERROR       mdns: mdns lookup error: failed to bind to any unicast udp port asm_amd64.s:2232
11:56:21.774 ERROR       mdns: mdns lookup error: failed to bind to any unicast udp port asm_amd64.s:2232
11:56:41.775 ERROR       mdns: mdns lookup error: failed to bind to any multicast udp port asm_amd64.s:2232
11:56:48.514 ERROR core/serve: Path Resolve error: context deadline exceeded gateway_handler.go:436
11:56:51.774 ERROR       mdns: mdns lookup error: failed to bind to any unicast udp port asm_amd64.s:2232
11:57:11.776 ERROR       mdns: mdns lookup error: failed to bind to any unicast udp port asm_amd64.s:2232
11:57:21.774 ERROR       mdns: mdns lookup error: failed to bind to any multicast udp port asm_amd64.s:2232
11:57:31.774 ERROR       mdns: mdns lookup error: failed to bind to any multicast udp port asm_amd64.s:2232
11:57:41.774 ERROR       mdns: mdns lookup error: failed to bind to any multicast udp port asm_amd64.s:2232
11:57:51.775 ERROR       mdns: mdns lookup error: failed to bind to any multicast udp port asm_amd64.s:2232
11:58:01.775 ERROR       mdns: mdns lookup error: failed to bind to any unicast udp port asm_amd64.s:2232
11:58:31.774 ERROR       mdns: mdns lookup error: failed to bind to any unicast udp port asm_amd64.s:2232
11:59:21.774 ERROR       mdns: mdns lookup error: failed to bind to any unicast udp port asm_amd64.s:2232
11:59:31.774 ERROR       mdns: mdns lookup error: failed to bind to any unicast udp port asm_amd64.s:2232
12:00:31.774 ERROR       mdns: mdns lookup error: failed to bind to any unicast udp port asm_amd64.s:2232
12:01:01.774 ERROR       mdns: mdns lookup error: failed to bind to any unicast udp port asm_amd64.s:2232
12:01:11.774 ERROR       mdns: mdns lookup error: failed to bind to any multicast udp port asm_amd64.s:2232
12:01:31.774 ERROR       mdns: mdns lookup error: failed to bind to any unicast udp port asm_amd64.s:2232
12:02:43.218 ERROR core/serve: Path Resolve error: context deadline exceeded gateway_handler.go:436
12:03:43.220 ERROR core/serve: Path Resolve error: context deadline exceeded gateway_handler.go:436
```

I did swarm peers and after about 30mins I tried it again and it seemed to work. Just wanted to report in case it was helpful.

**Comment by opn** [comment]
Would like to check that this is not an issue with recent builds. I’m on 0.3.2 - should I be good to upgrade :)

**Comment by whyrusleeping** [comment]
@MrChrisJ Interesting, My guess is that you and your colleagues nodes failed to connect to eachother due to NAT traversal failing, or an otherwise restrictive firewall.

@opn you shouldnt run into any issues upgrading

**Comment by diasdavid** [comment]
This should not be a issue in the latest release, let us know if you encounter any problems still.

Btw, `webui` has been receiving a lot of love recently https://github.com/ipfs/webui/commits/master , but still needs some help https://github.com/ipfs/webui/issues :)
